internal/services/containers/kubernetes_flux_configuration_resource.go

```diff
@@ -549,6 +550,16 @@ func (r KubernetesFluxConfigurationResource) Arguments() map[string]*pluginsdk.S
 						ConflictsWith: []string{"git_repository.0.https_user", "git_repository.0.ssh_private_key_base64", "git_repository.0.ssh_known_hosts_base64"},
 					},
 
+					"provider": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+						Default:  "generic",
+						ValidateFunc: validation.StringInSlice([]string{
+							"generic",
```

Comment:
Please use the constants defined in the sdk: `ProviderTypeGeneric` and `ProviderTypeAzure`